lms/services/group_set.py

```diff
@@ -0,0 +1,44 @@
+from typing import TypedDict
+
+from lms.models.group_set import LMSGroupSet
+
+
+class GroupSetDict(TypedDict):
+    """
+    Group sets are a collection of student groups.
+
+    We store them in Course.extra
+    """
+
+    id: str
+    name: str
+
+
+class GroupSetService:
```

Comment:
Note that's there's no read method here yet. To tackle those we have to finish the write side first.